### PR TITLE
Make locks work with bearer tokens

### PR DESCRIPTION
**Pull request by paulgb**
If a backend has a bearer token and a lock, and we send a spawn request with the same lock, the returned result now includes the bearer token.

This works by attaching the bearer token to the drone assignment message.

### CODE REVIEW

File: dev/tests/scheduler.rs [hunk 1]
```diff
@@ -452,3 +452,49 @@ async fn schedule_request_lock() {
...
+            drone_id: drone_id.clone(),
+            drone_version: PLANE_VERSION.to_string(),
+            ready: true,
+            state: DroneState::Ready,
+            running_backends: None,
+        })
+        .await
+        .unwrap();
+
+    sleep(Duration::from_millis(100)).await;
```
Comment (pretentious7) [inline dev/tests/scheduler.rs:479]: what condition is this sleep to ensure?

**Review by pretentious7 — APPROVED** [review]
LGTM!

File: core/src/messages/state.rs [hunk 2]
```diff
@@ -86,6 +86,7 @@ pub enum BackendMessageType {
     Assignment {
         drone: DroneId,
         lock: Option<String>,
+        bearer_token: Option<String>,
```
[the remark this answers is not in the document]
Comment (paulgb) [reply core/src/messages/state.rs:89]: My thinking is that any UTF-8 string is valid as far as Plane is concerned. It is only used for comparisons.

File: dev/tests/scheduler.rs (hunk 1 above)
Comment (paulgb) [reply dev/tests/scheduler.rs:479]: It was in the test I copied, I've forgotten at this point. I agree that we should go back and remove sleeps, but I'd rather merge with some sleeps in test for now and then clean up later.

**Pull request merged**